apis/python/src/tiledbsoma/_geometry_dataframe.py

```diff
@@ -244,16 +520,232 @@ def coordinate_space(self, value: CoordinateSpace) -> None:
+            )
+        dtype = schema.field(index_column_name).type
+        if not pa.types.is_dictionary(dtype) and dtype not in [
+            pa.int8(),
+            pa.uint8(),
+            pa.int16(),
+            pa.uint16(),
+            pa.int32(),
+            pa.uint32(),
+            pa.int64(),
+            pa.uint64(),
+            pa.float32(),
+            pa.float64(),
+            pa.binary(),
+            pa.large_binary(),
+            pa.string(),
+            pa.large_string(),
+            pa.timestamp("s"),
+            pa.timestamp("ms"),
+            pa.timestamp("us"),
+            pa.timestamp("ns"),
+        ]:
+            raise TypeError(
+                f"Unsupported index type {schema.field(index_column_name).type}"
+            )
+
+    return schema
+
+
+def _spatial_domain(
+    spatial_column_names: Sequence[str],
+    domain: AxisDomain,
+    platform_config: options.PlatformConfig,
+) -> Tuple[Sequence[pa.Field], Mapping[str, Any]]:
+    spatial_column_schema = []
+    spatial_column_data = {}
+
+    if domain is None:
+        spatial_domain: Domain = tuple(None for _ in spatial_column_names)
+    else:
+        spatial_domain = domain
+
+    for spatial_column_name, slot_spatial_domain in zip(
+        spatial_column_names, spatial_domain
+    ):
+        pa_field = pa.field(spatial_column_name, pa.float64())
+        dtype = _arrow_types.tiledb_type_from_arrow_type(
+            pa_field.type, is_indexed_column=True
+        )
+
+        (slot_core_current_domain, saturated_cd) = _fill_out_slot_soma_domain(
+            slot_spatial_domain, False, spatial_column_name, pa_field.type, dtype
+        )
+        (slot_core_max_domain, saturated_md) = _fill_out_slot_soma_domain(
+            None, True, spatial_column_name, pa_field.type, dtype
+        )
+
+        extent = _find_extent_for_domain(
+            spatial_column_name,
+            TileDBCreateOptions.from_platform_config(platform_config),
+            dtype,
+            slot_core_current_domain,
+        )
+
+        # Necessary to avoid core array-creation error "Reduce domain max by
+        # 1 tile extent to allow for expansion."
+        slot_core_current_domain = _revise_domain_for_extent(
+            slot_core_current_domain, extent, saturated_cd
+        )
+        slot_core_max_domain = _revise_domain_for_extent(
+            slot_core_max_domain, extent, saturated_md
+        )
+
+        # Here is our Arrow data API for communicating schema info between
+        # Python/R and C++ libtiledbsoma:
+        #
+        # [0] core max domain lo
+        # [1] core max domain hi
+        # [2] core extent parameter
+        # If present, these next two signal to use the current-domain feature:
+        # [3] core current domain lo
+        # [4] core current domain hi
+
+        spatial_column_schema.append(pa_field)
+        if NEW_SHAPE_FEATURE_FLAG_ENABLED:
```

Comment:
This flag has been removed as of https://github.com/single-cell-data/TileDB-SOMA/pull/3301. Will need to rebase with `main`.